lib/index.js

```diff
@@ -153,14 +159,32 @@ internals.initialize = function (server, next) {
 
 internals.stop = function (server, next) {
 
-    const collector = internals.state(server.root).collector;
+    const rootCollector = internals.state(server.root).collector;
 
     // Do not teardown if specifically asked not to
-    if (collector.teardownOnStop === false) {
+    if (rootCollector.teardownOnStop === false) {
         return next();
     }
 
-    return server.root.knex().destroy(next);
+    const knexConnections = [];
+
+    Object.keys(rootCollector.knexGroups).forEach((key) => {
+
+        knexConnections.push(rootCollector.knexGroups[key].knex);
+    });
+
+    // destroy() all knex instances
+    Items.parallel(knexConnections, (item, nxt) => {
+
+        item.destroy(() => {
+
+            nxt();
+        });
+    },
+    (_) => {
```

Comment:
1. You don't want to throw errors in async functions unless something completely unrecoverable has occurred.
2. You don't need to destroy the root knex connection twice.

```js
Items.parallel(knexConnections, (item, nxt) => item.destroy(nxt), next);
```